### PR TITLE
support default values in c# methods

### PR DESCRIPTION
**Pull request by bzbetty**
believe this makes default values in mutaitons work

eg

```

        [GraphQLMutation]

        public int DefaultValueTest(int valueWithDefault = 8)
        {
            return valueWithDefault;
        }
```

### CODE REVIEW

**Comment by bzbetty** [comment]
Think this is failing due to another merge. Gets the same Reuse slot error (which seem familiar to me, something about trying to add an Expression type to graphql?)

**Comment by lukemurray** [comment]
Can we add a test/check that the schema/introspection is generating the default values too?

**Comment by bzbetty** [comment]
added in tests/code for sdl/introspection default values

**Comment by bzbetty** [comment]
added a couple more SDL default values checks which requires some form of serialisation, used a json serialiser but it's not quite right, so need another method.

eg
`            schema.Query().AddField("user", new { user = new UserInputObject() { Name = "Steve" } }, (ctx, param) => ctx.Users.FirstOrDefault(), "Return a user by ID");
`
should become

`            Assert.Contains("user(user: userInput = { name:\"Steve\"}): User", schema.ToGraphQLSchemaString());
`
but is currently

`            Assert.Contains("user(user: userInput = { \"name\":\"Steve\"}): User", schema.ToGraphQLSchemaString());
`
which isn't valid SDL.

seems to work for enums, integers, booleans, and empty arrays though

File: src/EntityGraphQL/Schema/SchemaGenerator.cs [hunk 1]
```diff
@@ -1,187 +1,206 @@
...
-                if (!string.IsNullOrEmpty(field.Description))
-                    sb.AppendLine($"\t\"\"\"{EscapeString(field.Description)}\"\"\"");
-                sb.AppendLine($"\t{schema.SchemaFieldNamer(field.Name)}{GetGqlArgs(schema, field)}: {field.ReturnType.GqlTypeForReturnOrArgument}{GetDirectives(field.Directives)}");
-            }
-            sb.AppendLine("}");
-
-            return sb.ToString();
-        }
-    }
+using System;
```
Comment (lukemurray) [inline src/EntityGraphQL/Schema/SchemaGenerator.cs:1]: Why does it show this file 100% changed? Makes it hard to see what did change!

Any highlights?

Comment (bzbetty) [reply src/EntityGraphQL/Schema/SchemaGenerator.cs:1]: curious, i must have accidentally formatted it differently. will see if I can clean it up.

there should only be about 2 lines different where it serializes the default value, that said i need to write a custom serializer i think.

Comment (bzbetty) [reply src/EntityGraphQL/Schema/SchemaGenerator.cs]: image

and GetArgDefaultValue method is entirely new

**Comment by lukemurray** [comment]
My most hated test too :P Need to look at changing it some how..

Thanks though

**Pull request merged**